google/cloud/security/enforcer/gce_firewall_enforcer.py

```diff
@@ -486,6 +486,9 @@ def add_rules(self, rules, network_name=None):
           InvalidFirewallRuleError: One or more rules failed validation.
         """
         for rule in rules:
```

Comment:
If I put this in add_rule I'd still check this on line one, so why not check here? This would prevent entering the function unnecessarily.